.circleci/do-firesim-build.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# create the different verilator builds
+# argument is the make command string
+
+# turn echo on and error on earliest command
+set -ex
+
+# get shared variables
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+source $SCRIPT_DIR/defaults.sh
+
+# call clean on exit
+trap clean EXIT
+
+cd $LOCAL_CHIPYARD_DIR
+./scripts/init-submodules-no-riscv-tools.sh
+cd sims/firesim/sim/midas && git submodule update --init
+
+# set stricthostkeychecking to no (must happen before rsync)
+run "echo \"Ping $SERVER\""
+
+clean
+
+# copy over riscv/esp-tools, verilator, and chipyard to remote
+run "mkdir -p $REMOTE_CHIPYARD_DIR"
+run "mkdir -p $REMOTE_VERILATOR_DIR"
+copy $LOCAL_CHIPYARD_DIR/ $SERVER:$REMOTE_CHIPYARD_DIR
+copy $LOCAL_VERILATOR_DIR/ $SERVER:$REMOTE_VERILATOR_DIR
+
+run "cp -r ~/.ivy2 $REMOTE_WORK_DIR"
```

Comment:
Why is this being done... shouldn't the `.ivy` and `.sbt` directories be created from scratch when you invoke `sbt`?

Reply:
The problem is that would cause all the external dependencies to be downloaded again. I'm trying to avoid doing unnecessary work.